### ISSUE TITLE
[pixi] ReferenceError: window is not defined in Node.js

### ISSUE THREAD

**Issue by vadimgoncharov** [EsotericSoftware/spine-runtimes#2409]
**Problem:**
https://github.com/EsotericSoftware/spine-runtimes/blob/fa4310b4d3f44f5b5c9cad050f46dfe19f1bb831/spine-ts/spine-pixi/src/require-shim.ts#L35

In `require-shim.ts` we have this code:
```
if (window.PIXI) {
	let prevRequire = window.require;
	window.require = (x: string) => {
		if (prevRequire) return prevRequire(x);
		else if (x.startsWith("@pixi/")) return window.PIXI;
	}
}
```

Because `window` doesn't exist in Node.js we have problem in Node.js when we try to use testing libraries like vitest. 

We've got this error:
```
ReferenceError: window is not defined
 ❯ ../node_modules/@esotericsoftware/spine-pixi/src/require-shim.ts:35:1
 ❯ ../node_modules/@esotericsoftware/spine-pixi/dist/index.js:3:31
```
<img width="950" alt="image" src="https://github.com/EsotericSoftware/spine-runtimes/assets/357696/059b6cdd-bcec-484f-8948-1650541886cd">

**Solution:**
Add `typeof window !== 'undefined'`:
```
if (typeof window !== 'undefined' && window.PIXI) {
	let prevRequire = window.require;
	window.require = (x: string) => {
		if (prevRequire) return prevRequire(x);
		else if (x.startsWith("@pixi/")) return window.PIXI;
	}
}
```

**Comment by badlogic** [comment]
Oh, that's a use case I've never considered. How are you testing Spine animations in a headless environment like Node.js?

**Comment by badlogic** [comment]
I've just released new packages with this fix to NPM, 4.1.44 and 4.2.25. Thanks for your PR!

**Comment by vadimgoncharov** [comment]
> Oh, that's a use case I've never considered. How are you testing Spine animations in a headless environment like Node.js?

We have code like this :)
```
import {Spine} from '@esotericsoftware/spine-pixi'

class GameService {
  constructor() {
    this.character = this.createSpine('characterAltas', 'characterJson')
  }

  public playAnimationByUserInput(input: string): void {
     if (input === '...') {
       this.playAnimationX();
     } else if (...) {
       this.playAnimationY();
     } else {...}
  }

  private playAnimationX() {
    const animation = this.character.state.setAnimation(0, 'animationNameX', false);
    // ...
  }

  private playAnimationY() {
    const animation = this.character.state.setAnimation(0, 'animationNameY', false);
    // ...
  }
 
  private createSpine(atlasName: string, jsonName: string): Spine {
     // creating spine
     // ...
    return spine
  }
}
```

We want to test:
- if "userInput === X" then "playAnimationX" would be triggered
- if "userInput === Y" then "playAnimationY" would be triggered

```
import {GameService} from './game.service.ts'

describe('GameService', () => {
  it('triggers "playAnimationX"', () => {
    const game = new GameService()
    const spy = attachSpyOn(game, 'playAnimationX')
    game.playAnimationByUserInput('X')
    expect(spy).toHaveBeenCalled()
  })
})
```

So if we run this test on NodeJS it will import `game.service.ts` which will import `@esotericsoftware/spine-pixi` which will import `require-shim.ts` which will triggers undefined error.

We solved problem by incapsulating `@esotericsoftware/spine-pixi'` in `spine.service.ts` and using dependency injection, so in NodeJS we don't import `@esotericsoftware/spine-pixi` by importing mocked `spine.service.mock.ts`, but it took a lot of time :)

Thanks for new release! 🔥
